live-examples/js-examples/string/string-iterator.html

```diff
@@ -0,0 +1,12 @@
+<pre>
+<code id="static-js">var str = "The quick red fox jumped over the lazy dog's back.";
+
+  var iterator = str[Symbol.iterator]();
+
+  do {
+    var theChar = iterator.next();
+    // output each character in the string, one per line
+    console.log(theChar);
+  } while(!theChar.done)  
```

Comment:
With this logic the last time round will output `Object { value: undefined, done: true }` (or just `undefined` if you call `theChar.value`). I'm not sure if this is intentional.

If not you could avoid it by doing something like:

```
let theChar = iterator.next();

while (!theChar.done) {
  // output each character in the string, one per line
  console.log(theChar.value);
  theChar = iterator.next();
} 
```